core/code/toolbox.js

```diff
@@ -0,0 +1,245 @@
+   * @returns {boolean} True if the button is successfully updated, false otherwise.
+   *
+   * @example
+   * const isUpdated = IITC.toolbox.updateButton(buttonId, { label: 'Updated Button', action: () => console.log('New Action') });
+   */
+  updateButton(buttonId, newButtonArgs, options = { legacy: false }) {
+    if (options.legacy) {
+      buttonId = 'legacy-' + buttonId;
+    }
+
+    if (this.buttons[buttonId]) {
+      // Обновляем только указанные ключи в объекте кнопки
+      Object.assign(this.buttons[buttonId], newButtonArgs);
+      this._renderButton(buttonId);
+      this._applySort();
+      return true;
+    } else {
+      console.warn(`Button with ID ${buttonId} not found.`);
+      return false;
+    }
+  },
+
+  /**
+   * Removes a button from the toolbox.
+   *
+   * @param {string} buttonId - The ID of the button to remove.
+   * @returns {boolean} True if the button is successfully removed, false otherwise.
+   *
+   * @example
+   * const isRemoved = IITC.toolbox.removeButton(buttonId);
+   */
+  removeButton(buttonId) {
+    if (this.buttons[buttonId]) {
+      delete this.buttons[buttonId];
+      const buttonElement = document.getElementById(buttonId);
+      if (buttonElement) {
+        buttonElement.remove();
+      }
+      this._applySort();
+      return true;
+    } else {
+      console.warn(`Button with ID ${buttonId} not found for removal.`);
+      return false;
+    }
+  },
+
+  /**
+   * Internal method to render a button.
+   *
+   * @private
+   * @param {string} buttonId - The ID of the button to render.
+   */
+  _renderButton(buttonId) {
+    const buttonData = this.buttons[buttonId];
+    if (!buttonData) return; // The button with the given ID was not found
+
+    let buttonElement = document.getElementById(buttonId) || document.createElement('a');
+    buttonElement.id = buttonId;
+    buttonElement.textContent = buttonData.label;
+    buttonElement.onclick = buttonData.action;
+
+    if (buttonData.class) buttonElement.className = buttonData.class;
+    if (buttonData.title) buttonElement.title = buttonData.title;
+    if (buttonData.access_key) buttonElement.accessKey = buttonData.access_key;
+    if (buttonData.mouseover) buttonElement.mouseover = buttonData.mouseover;
+
+    if (buttonData.icon) {
+      const iconHTML = `<i class="fa ${buttonData.icon}"></i>`;
+      buttonElement.innerHTML = iconHTML + buttonElement.innerHTML;
+    }
+
+    const toolbox_component = document.querySelector('#toolbox_component');
+    if (!document.getElementById(buttonId)) {
+      toolbox_component.appendChild(buttonElement);
+    }
+  },
+
+  /**
+   * Internal method to apply sorting to the buttons.
+   *
+   * @private
+   */
+  _applySort() {
+    const toolbox_component = document.querySelector('#toolbox_component');
+    const buttonElements = Array.from(toolbox_component.children);
+
+    buttonElements.sort((a, b) => this.sortMethod(this.buttons[a.id], this.buttons[b.id]));
```

Comment:
imho we should do smth like this to avoid broken sortMethods from 3rd party plugins. Console log should be enough, but we could also add some warning to toolbox that custom sorting was attemted but failed 🤷‍♀️
```suggestion
    try {
      buttonElements.sort((a, b) => this.sortMethod(this.buttons[a.id], this.buttons[b.id]));
    } catch (e) {
      console.error("Sorting function produced error", e);
      buttonElements.sort((a, b) => this._defaultSortMethod(this.buttons[a.id], this.buttons[b.id]));
    }
```